earth_enterprise/src/common/khxml/khxml.cpp

```diff
@@ -433,17 +601,42 @@ DestroyDocument(khxml::DOMDocument *doc) throw()
     retval = true;
   } catch (...) {
   }
+  if (terminateCache)
+  {
+        khLockGuard guard(checkTermLock);
+        terminateGuard::instance().removeObj();
+        if (terminateGuard::instance().isLocked() &&
+            !terminateGuard::instance().size())
```

Comment:
Minor: I prefer `terminateGuard::instance().size() == 0` - I think it's a bit clearer.